pkg/azureclients/fileclient/azure_fileclient.go

```diff
@@ -176,10 +180,30 @@ func (c *Client) ResizeFileShare(ctx context.Context, resourceGroupName, account
 }
 
 // GetFileShare gets a file share
-func (c *Client) GetFileShare(ctx context.Context, resourceGroupName, accountName, name string) (storage.FileShare, error) {
+// xMsSnapshot - optional, used to retrieve properties of a snapshot.
+// It is a DateTime value that uniquely identifies the share snapshot. e.g. "2017-05-10T17:52:33.9551861Z"
+func (c *Client) GetFileShare(ctx context.Context, resourceGroupName, accountName, name, xMsSnapshot string) (storage.FileShare, error) {
 	mc := metrics.NewMetricContext("file_shares", "get", resourceGroupName, c.subscriptionID, "")
 
-	result, err := c.fileSharesClient.Get(ctx, resourceGroupName, accountName, name, "stats", "")
+	result, err := c.fileSharesClient.Get(ctx, resourceGroupName, accountName, name, "stats", xMsSnapshot)
+	var rerr *retry.Error
+	if err != nil {
+		rerr = &retry.Error{
+			RawError: err,
+		}
+	}
+	mc.Observe(rerr)
+
+	return result, err
+}
+
+// ListFileShare gets a file share list
+// expand - optional, used to expand the properties within share's properties. Valid values are: deleted,
+// snapshots. Should be passed as a string with delimiter ','
+func (c *Client) ListFileShare(ctx context.Context, resourceGroupName, accountName, filter, expand string) (storage.FileShareItemsPage, error) {
```

Comment:
this func may only return part of file share snapshots, you could refer to this function to return all snapshots:
https://github.com/kubernetes-sigs/cloud-provider-azure/blob/b0d976d580d9a0252febacc7bedd9f6def79e817/pkg/azureclients/snapshotclient/azure_snapshotclient.go#L328-L342

Reply:
ok, should we use retry.Error for output or error is ok?